### ISSUE TITLE
cannot read property env of undefined

### ISSUE THREAD

**Issue by robclancy** [ef4/ember-auto-import#24]
Happening on this line https://github.com/ef4/ember-auto-import/blob/2f2dec795ed277c0ae13aa210698bf0a6a32a860/index.js#L45

```
ERROR Summary:

  - broccoliBuilderErrorStack: [undefined]
  - codeFrame: [undefined]
  - errorMessage: Cannot read property 'env' of undefined
  - errorType: [undefined]
  - location:
    - column: [undefined]
    - file: [undefined]
    - line: [undefined]
  - message: Cannot read property 'env' of undefined
  - name: TypeError
  - nodeAnnotation: [undefined]
  - nodeName: [undefined]
  - originalErrorMessage: [undefined]
  - stack: TypeError: Cannot read property 'env' of undefined
    at Class.included (/home/robbo/projects/postedin/marketplace/node_modules/ember-auto-import/index.js:43:48)
    at addons.reduce (/home/robbo/projects/postedin/marketplace/node_modules/ember-cli/lib/models/addon.js:381:26)
    at Array.reduce (native)
    at Class.eachAddonInvoke (/home/robbo/projects/postedin/marketplace/node_modules/ember-cli/lib/models/addon.js:378:24)
    at Class.included (/home/robbo/projects/postedin/marketplace/node_modules/ember-cli/lib/models/addon.js:648:10)
    at Class.superWrapper [as included] (/home/robbo/projects/postedin/marketplace/node_modules/ember-cli/node_modules/core-object/lib/assign-properties.js:34:20)
    at addons.reduce (/home/robbo/projects/postedin/marketplace/node_modules/ember-cli/lib/models/addon.js:381:26)
    at Array.reduce (native)
    at Class.eachAddonInvoke (/home/robbo/projects/postedin/marketplace/node_modules/ember-cli/lib/models/addon.js:378:24)
    at Class.included (/home/robbo/projects/postedin/marketplace/node_modules/ember-cli/lib/models/addon.js:648:10)
```

**Comment by sfbwalder** [comment]
I have the same error when a parent addon has a child addon that uses ember-auto-import, and the child addon is defined as a dependency (not devDependency) of the parent addon.

```
// parent addon package.json config that does not work
{
  "name": "parent-addon",
  "dependencies": {
    "ember-cli-babel": "^6.6.0",
    "child-addon": "^0.0.1" 
  }
}
```
```
// child addon package.json
{
  "name": "child-addon",
  "version": "0.0.1",
  "dependencies": {
    "ember-auto-import": "^0.2.2",
    "ember-cli-babel": "^6.6.0",
    "urijs": "^1.19.1"
  }
}
```

If I move the child addon to the devDependencies of the parent addon, it works.

```
// parent addon package.json config that does not work
{
  "name": "parent-addon",
  "dependencies": {
    "ember-cli-babel": "^6.6.0",
  },
  "devDependencies": {
    "child-addon": "^0.0.1"
  },
}
```

Is this the expected behavior?

**Comment by ef4** [comment]
Thanks, that makes sense. When an addon runs as an addon (as opposed to its own dummy app), it only uses dependencies, not devDependencies (which follows from the way NPM works).

It sounds like the bug here is in how we identify the environment from within second-level addons.

**Comment by ef4** [comment]
Released in 0.2.6

**Comment by robclancy** [comment]
Nice time to give this a go again.
